src/gen_rules.ml

```diff
@@ -28,30 +28,53 @@ module Gen(P : Params) = struct
      +-----------------------------------------------------------------+ *)
 
   module Eval_modules = Ordered_set_lang.Make(struct
-      type t = Module.t
-      let name = Module.name
+      type t = Loc.t * ((Module.t, string) result)
```

Comment:
Why do we need to change this type?

It seems to me that it's enough to only keep track of non-existent modules in `fake_modules`. We need to report an error for modules that are in `fake_modules` and `modules` and a warning for modules that  are in `fake_modules` but not in `modules`. Either way, the module we report or warn for are in `fake_modules`, so we know their location.

Reply:
I needed to change the return type because we now need to return something sensible out of `parse ~loc s` when the module is missing rather than raising. Also, the loc is used later to report actual errors since we now must delay those.

> We need to report an error for modules that are in fake_modules and modules

Isn't this why we need to save the loc? we use it later on to report these errors.

Reply:
well the loc is already saved in `fake_modules`

Reply:
Ah ok, fake_modules is a superset of all the missing modules we might error on so we might as well use that for the locs.